src/bika/lims/browser/worksheet/views/add_analyses.py

```diff
@@ -114,6 +115,12 @@ def __init__(self, context, request):
 
     def __call__(self):
         super(AddAnalysesView, self).__call__()
+        wst = self.context.getWorksheetTemplate()
+        if wst:
+            method = wst.getRawRestrictToMethod()
+            # restrict the available analysis services by method
+            if method:
+                self.contentFilter["getServiceUID"] = getServiceUidsByMethod(method)
```

Comment:
Better move all this inside the function `update` so the filter applies from the start. Note the `update` function is called by the superclass `ListingView` in `__call__`: https://github.com/senaite/senaite.app.listing/blob/2.x/src/senaite/app/listing/view.py#L220

Reply:
Also take into account that `getServiceUidsByMethod` may return an empty list

Reply:
is it right, because if method not empty in the worksheet and not exists samples with this method than nothing show(empty list)
or i wrong?